tests/core/usage_statistics/test_usage_statistics_data_context.py

```diff
@@ -0,0 +1,122 @@
+        if name.startswith("__") and name.endswith("__"):
+            continue
+
+        decorators = node.decorator_list
+        for decorator in decorators:
+            if isinstance(decorator, ast.Call):
+                if decorator.func.id == "usage_statistics_enabled_method":
+                    relevant_methods.append(name)
+                    break
+
+    return relevant_methods
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@pytest.mark.integration
+def test_all_relevant_context_methods_emit_usage_stats(
+    mock_emit: mock.MagicMock,
+    usage_stats_decorated_methods_on_abstract_data_context: list[str],
+    enable_usage_stats,  # Needs to be before context fixtures to ensure usage stats handlers are attached
+    in_memory_runtime_context: EphemeralDataContext,
+    empty_data_context: FileDataContext,
+    empty_data_context_in_cloud_mode: CloudDataContext,
+):
+    """
+    What does this test and why?
+
+    Ensures that all primary context types (EphemeralDataContext, FileDataContext, and CloudDataContext)
+    emit the same usage stats events.
+
+    This guards against the case where a child class overrides a method defined by AbstractDataContext
+    but forgets to add the decorator, resulting in us losing event data.
+    """
+
+    relevant_methods = usage_stats_decorated_methods_on_abstract_data_context
+    expected_events = (
+        UsageStatsEvents.DATA_CONTEXT_SAVE_EXPECTATION_SUITE,
+        UsageStatsEvents.DATA_CONTEXT_ADD_DATASOURCE,
+        UsageStatsEvents.DATA_CONTEXT_RUN_CHECKPOINT,
+        UsageStatsEvents.DATA_CONTEXT_GET_BATCH_LIST,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_WITH_DYNAMIC_ARGUMENTS,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_ON_DATA,
+        UsageStatsEvents.DATA_CONTEXT_RUN_VALIDATION_OPERATOR,
+        UsageStatsEvents.DATA_CONTEXT_OPEN_DATA_DOCS,
+        UsageStatsEvents.DATA_CONTEXT_BUILD_DATA_DOCS,
+    )
+    assert len(relevant_methods) == len(
+        expected_events
+    ), "Please update the `expected_events` list to account for all usage stats decorated methods in AbstractDataContext"
+
+    contexts = (
+        in_memory_runtime_context,
+        empty_data_context,
+        empty_data_context_in_cloud_mode,
+    )
+
+    for context in contexts:
+        for method_name, expected_event in zip(relevant_methods, expected_events):
+            logger.info(f"Testing {context.__class__}.{method_name}")
+
+            # As we only care about the decorator and not the underlying method being decorated,
+            # we use the following try/except pattern. All method calls will generally fail due
+            # to having no input args but we still manage to trigger our target decorator.
+            try:
+                method = getattr(context, method_name)
+                method()
+            except Exception:
+                pass
```

Comment:
Does this feel bad? Maybe a little but it works pretty well? We only really care about triggering the event